digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/ServicePointBusStopDao.scala

```diff
@@ -0,0 +1,262 @@
+import slick.jdbc.{GetResult, PositionedResult, StaticQuery => Q}
+import fi.liikennevirasto.digiroad2.asset.PropertyTypes._
+import fi.liikennevirasto.digiroad2.asset.{MassTransitStopValidityPeriod, _}
+import fi.liikennevirasto.digiroad2.dao.Queries._
+import fi.liikennevirasto.digiroad2.service.pointasset.masstransitstop.MassTransitStopOperations
+import org.joda.time.{DateTime, LocalDate}
+import org.slf4j.LoggerFactory
+import fi.liikennevirasto.digiroad2.dao.Queries._
+import org.joda.time.DateTime
+import slick.driver.JdbcDriver.backend.Database
+import Database.dynamicSession
+import fi.liikennevirasto.digiroad2.Point
+import slick.jdbc.StaticQuery.interpolation
+import slick.jdbc.{GetResult, PositionedResult}
+import scala.language.reflectiveCalls
+import fi.liikennevirasto.digiroad2.Point
+import fi.liikennevirasto.digiroad2.asset.{LinkGeomSource, Modification, Property}
+import fi.liikennevirasto.digiroad2.dao.Queries.PropertyRow
+import fi.liikennevirasto.digiroad2.service.pointasset.masstransitstop.PersistedMassTransitStop
+import org.joda.time.LocalDate
+import org.slf4j.LoggerFactory
+import slick.jdbc.StaticQuery.interpolation
+import slick.jdbc.{GetResult, PositionedParameters, PositionedResult, SetParameter, StaticQuery => Q}
+import slick.driver.JdbcDriver.backend.Database
+import Database.dynamicSession
+import fi.liikennevirasto.digiroad2._
+import fi.liikennevirasto.digiroad2.asset.PropertyTypes._
+import fi.liikennevirasto.digiroad2.asset.{MassTransitStopValidityPeriod, _}
+import fi.liikennevirasto.digiroad2.dao.Queries._
+import fi.liikennevirasto.digiroad2.model.LRMPosition
+import fi.liikennevirasto.digiroad2.oracle.OracleDatabase
+import fi.liikennevirasto.digiroad2.service.pointasset.masstransitstop.{LightGeometryMassTransitStop, MassTransitStopOperations, MassTransitStopRow, PersistedMassTransitStop}
+import org.joda.time.{DateTime, Interval, LocalDate}
+import org.joda.time.format.ISODateTimeFormat
+import org.slf4j.LoggerFactory
+import scala.language.reflectiveCalls
+
+case class ServicePoint(id: Long, nationalId: Long, stopTypes: Seq[Int],
+                        municipalityCode: Int, lon: Double, lat: Double,
+                        created: Modification, modified: Modification, propertyData: Seq[Property])
+
+case class ServicePointRow(id: Long, externalId: Long, assetTypeId: Long, point: Option[Point],
+                           validFrom: Option[LocalDate], validTo: Option[LocalDate], property: PropertyRow,
+                           created: Modification, modified: Modification, municipalityCode: Int)
+
+class ServicePointBusStopDao {
+  val logger = LoggerFactory.getLogger(getClass)
+  def typeId: Int = MassTransitStopAsset.typeId
+  val idField = "external_id" //???
+
+  private implicit val getLocalDate = new GetResult[Option[LocalDate]] {
+    def apply(r: PositionedResult) : Option[LocalDate] = {
+      r.nextDateOption().map(new LocalDate(_))
+    }
+  }
```

Comment:
Code copied from MassTransitStopDao, let's try to move the original code to one place where anyone can use it since now it's used for more than one file.

Reply:
@marquesfm This part was removed since this dao now extends from the masstransitstop one